src/browser-installer/registry/index.ts

```diff
@@ -12,204 +13,317 @@ import {
 
-        default:
-            return null;
+                return executablePath;
+            })
+            .catch(err => {
+                progressBar?.stop();
+
+                throw err;
+            });
+
+        this.registry.binaries[registryKey] ||= {};
+        this.registry.binaries[registryKey][version] = installPromise;
+
+        return installPromise;
     }
 
-    const buildIds = getBinaryVersions(browserName, platform);
-    const suitableBuildIds = buildIds.filter(buildId => buildId.startsWith(buildPrefix));
+    public async installOsPackages(
+        osName: OsName,
+        version: OsVersion,
+        installFn: (downloadProgressCallback: DownloadProgressCallback) => Promise<string>,
+    ): Promise<string> {
+        const registryKey = getRegistryOsPackagesKey(osName, version);
 
-    if (!suitableBuildIds.length) {
-        return null;
+        if (this.hasOsPackages(osName, version)) {
+            return this.getOsPackagesPath(osName, version);
+        }
+
+        browserInstallerDebug(`installing os packages for '${osName}@${version}'`);
+
+        logDownloadingOsPackagesWarningOnce(osName);
+
+        const progressBar = await getCliProgressBar();
+
+        const downloadProgressCallback = progressBar.register(`extra packages for ${osName}`, version);
+
+        const installPromise = installFn(downloadProgressCallback)
+            .then(packagesPath => {
+                this.addOsPackageToRegistry(osName, version, packagesPath);
+
+                return packagesPath;
+            })
+            .catch(err => {
+                progressBar.stop();
+
+                throw err;
+            });
+
+        this.registry.osPackages[registryKey] = installPromise;
+
+        return installPromise;
     }
 
-    const firefoxVersionComparator = (a: string, b: string): number => {
-        a = a.slice(a.indexOf("_") + 1);
-        b = b.slice(b.indexOf("_") + 1);
+    private readRegistry(): RegistryFileContents {
+        const registry: RegistryFileContents = {
+            binaries: {} as Record<BinaryKey, VersionToPathMap>,
+            osPackages: {} as Record<OsPackagesKey, string>,
+            meta: { version: 1 },
+        };
 
-        // Firefox has versions like "stable_131.0a1" and "stable_129.0b9"
-        // Parsing raw numbers as hex values is needed in order to distinguish "129.0b9" and "129.0b7" for example
-        return parseInt(a.replace(".", ""), 16) - parseInt(b.replace(".", ""), 16);
-    };
+        let fsData: Record<string, unknown>;
 
-    const comparator = browserName === Browser.FIREFOX ? firefoxVersionComparator : semverVersionsComparator;
-    const suitableBuildIdsSorted = suitableBuildIds.sort(comparator);
+        if (fs.existsSync(this.registryPath)) {
+            fsData = fs.readJSONSync(this.registryPath);
 
-    return suitableBuildIdsSorted[suitableBuildIdsSorted.length - 1];
-};
+            const isRegistryV0 = fsData && !fsData.meta;
```

Comment:
lets get rid of V0 support. we can try to install browsers and drivers again if the `meta` field is missing (with generating new registry)

Reply:
You mean, "delete V0 file"?